src/qmozview_p.cpp

```diff
@@ -1106,6 +1106,11 @@ void QMozViewPrivate::OnFirstPaint(int32_t aX, int32_t aY)
 #endif
     mIsPainted = true;
     mViewIface->firstPaint(aX, aY);
```

Comment:
This introduces possibility that domContent is not loaded before we got a painted frame (no sure how would I do that but). I'd rather follow ```painted``` property on the browser side.

Reply:
This was it https://github.com/sailfishos/sailfish-browser/pull/485 ( https://github.com/sailfishos/sailfish-browser/commit/0af8ad3563e8b70577de4bb2828f10537f7136ab ).

The issue was other way around that first paint is not always triggered and domContentLoaded is there as a final fallback. 

So adding connection to the firstPaint and extending guards (similarly also within side of the updateActiveTabRendered) should do the same.

```
connect(m_webPage.data(), &QOpenGLWebPage::firstPaint,
              this, &DeclarativeWebContainer::updateActiveTabRendered, Qt::UniqueConnection);
...
if (m_webPage->completed() && m_webPage->active() && (m_webPage->isPainted() || m_webPage->domContentLoaded())) {
    m_webPage->update();
}
```

Reply:
+
```
-    readyToPaint: resourceController.videoActive ? webView.visible && !resourceController.displayOff : webView.visible && webView.contentItem && webView.contentItem.domContentLoaded
+    readyToPaint: resourceController.videoActive ? webView.visible && !resourceController.displayOff : webView.visible && webView.contentItem
+                                                   && (webView.contentItem.domContentLoaded || webView.contentItem.painted)
```